contrib/mod_tls.c

```diff
@@ -2844,6 +2882,10 @@ static DH *tls_dh_cb(SSL *ssl, int is_export, int keylen) {
     }
   }
 
+  if (use_pkeylen) {
+    keylen = pkeylen;
```

Comment:
This addresses one part of the issue, where for a 2048-bit server cert, we _should_ have selected the 2048-bit DH, but instead we only used the unadjusted 1024-bit length provided by OpenSSL.